crates/core/src/buc/connection.rs

```diff
@@ -0,0 +1,44 @@
+use crate::{
+	cnf::{GLOBAL_BUCKET, GLOBAL_BUCKET_ENFORCED},
+	err::Error,
+};
+use dashmap::DashMap;
+use std::sync::Arc;
+
+use super::store::{prefixed::PrefixedStore, Key, ObjectStore};
+
+// Helper type to represent how bucket connections are persisted
+pub type BucketConnections = DashMap<(String, String, String), Arc<dyn ObjectStore>>;
+
+/// Connect to a global bucket, if one is configured
+/// If no global bucket is configured, the NoGlobalBucket error will be returned
+/// The key in the global bucket will be: `{ns}/{db}/{bu}`
+pub fn connect_global(ns: &str, db: &str, bu: &str) -> Result<Arc<dyn ObjectStore>, Error> {
+	// Obtain the URL for the global bucket
+	let Some(ref url) = *GLOBAL_BUCKET else {
+		return Err(Error::NoGlobalBucket);
+	};
+
+	// Connect to the global bucket
+	let global = connect(url, true, false)?;
+
+	// Create a prefixstore for the specified bucket
+	let key = Key::from(format!("/{ns}/{db}/{bu}"));
```

Comment:
Shall we use `src/key/database/bu.rs` so everything related to the bucket keys are built in one place?

Reply:
This is not a key in the KV store, but rather within a global bucket. 
Let's say that you have a global bucket configured, and then configure a bucket "test" inside a database, then a file pointer like `f"test:/a.txt"` will be stored under `ns/db/test/a.txt` inside that globally defined bucket.